### DISCUSSION TITLE
[Question/Fuature Request] HttpEndpoint multi-node supported?

### DISCUSSION THREAD

**Discussion by regan32** [elsa-workflows/elsa-core#4584] [Q&A]
When:
I have X nodes of  elsa-workflow behind nginx, and i want to handle webhooks by workflow. Node, on which workflow is running, should notify other nodes by some kind of signal(redis signal?) about new endpoint. But it's not happening, only one node is able to process request.

Question: 
1) Is it supported, or planned to be supported?
2) If it's already supported, what is wrong with my configuration?

                .AddElsa(options =>
                {
                    options
                        .UseEntityFrameworkPersistence(
                            ef => ef
                                .UsePostgreSql(postgresqlConnectionString!)
                                .EnableDetailedErrors(settings?.EnableDetailedErrors ?? false)
                                .EnableSensitiveDataLogging(settings?.EnableSensitiveDataLogging ?? false),
                            autoRunMigrations: autoRunMigrations)
                        .UseRedisCacheSignal()
                        .ConfigureDistributedLockProvider(o => o.UseRedisLockProvider())

                        // Custom services
                        .AddCustomTenantAccessor<ExecutionContextTenantAccessor>()
                        .UseTenantSignaler()

                        // Activities
                        .AddHttpActivities(configureHttpClient: (o) => { o.AddPolicyHandler(DefaultElsaPolicy); })
                        .AddActivity<WriteLog>()
                        .AddUserTaskActivities()
                        .AddEmailActivities()
                        .AddCommonTemporalActivities(timer => timer.UseQuartzProvider(registerQuartz: false))

                        // Serializer
                        .ConfigureJsonSerializer(
                            (_, s) =>
                            {
                                s.ContractResolver = new DefaultContractResolver
                                {
                                    NamingStrategy = new CamelCaseNamingStrategy
                                    {
                                        ProcessDictionaryKeys = false,
                                        ProcessExtensionDataNames = true,
                                        OverrideSpecifiedNames = false,
                                    },
                                };
                                s.Converters.Add(new DecimalJsonConverter());
                                s.DateParseHandling = Newtonsoft.Json.DateParseHandling.None;
                            });
                    var services = options.Services;
                    services.Remove(services.Single(d => d.ImplementationType?.FullName == "Elsa.Activities.Http.Parsers.Response.JTokenHttpResponseContentReader"));
                    services.AddSingleton<IHttpResponseContentReader, JTokenHttpResponseContentReader>();

                    configure?.Invoke(options);
                })
                .AddLiquidFilter<ExtendedJsonFilter>("json");

**Comment by sfmskywalker** [comment]
Hello @regan32,

I want to make sure I understand your question: are you saying that when you publish your workflow definition from a certain node, only that node can handle the HTTP requests? If that's the case, then yes, Elsa supports this scenario.

It's important to configure a service bus so Elsa can send events to other nodes. When you publish a workflow, an event is sent to the service bus, allowing other nodes to clear their local caches, like the caches for workflow blueprints and triggers.

You can find more information on this setup [here](https://v2.elsaworkflows.io/docs/hosting/distributed-hosting#service-bus-broker).
